spec/ddtrace/contrib/delayed_job/plugin_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'spec_helper'
+require 'ddtrace/contrib/delayed_job/plugin'
+
+require_relative 'app'
+
+SampleJob = Struct.new('SampleJob') { def perform; end }
+
+RSpec.describe Datadog::Contrib::DelayedJob::Plugin do
+  let(:pin) { Datadog::Pin.get_from(::Delayed::Worker) }
+  let(:tracer) { ::Datadog::Tracer.new(writer: FauxWriter.new) }
+
+  before do
+    Datadog::Contrib::DelayedJob::Patcher.patch
+
+    pin.tracer = tracer
+    Delayed::Worker.delay_jobs = false
+  end
+
+  describe 'instrumenting worker execution' do
+    let(:worker) { double(:worker, name: 'worker') }
+    before do
+      allow(tracer).to receive(:shutdown!).and_call_original
```

Comment:
Is this not a no-op? `allow` is a stub that does no assertions, and using `and_call_original`, its just passing through like it was never there.

If you'd like to assert that `shutdown!` is called, it should be changed to `expect(tracer)`. If you want it to simply skip side effects from `#shutdown!`, then you can remove `and_call_original`.

Reply:
Interesting, I've never seen `have_received` before, but that's nice.

But the question still remains, why stub with a no-op just to override with an expectation below?  Is it just [this implementation](https://relishapp.com/rspec/rspec-mocks/docs/basics/spies#spy-on-a-method-on-a-partial-double) of a spy?